src/state_machine/signer/mod.rs

```diff
@@ -451,6 +511,10 @@ impl<SignerType: SignerTrait> Signer<SignerType> {
             .iter()
             .cloned()
             .collect::<HashSet<u32>>();
+
+        self.dkg_private_begin_msg = Some(dkg_private_begin.clone());
```

Comment:
Is this DkgPrivateDistribute state really necessary as it seems to immediately move to DkgPrivateGather?

Reply:
Yeah it can go away, it's just there for consistency so all the states are modeled.  But in fairness all of the `Distribute` states are ephemeral, and end up switching immediately to the corresponding `Gather` states.  In some cases we go back through the `process` loop before switching, sometimes it's in a function call chain.  I think `DkgPrivateDistribute` is the only one that changes inside a single call.

Reply:
Yeah we could re-architect all of this and get rid of all the `Distribute` states in both signer and coordinators, but it seems cleaner to leave them in.  We'd have to redo the whole coordinator process state machine loop to get rid of it there, and if you recall that was extremely involved when we tried it before.